packages/manager/apps/pci-private-network/src/data/hooks/networks/useNetworks.ts

```diff
@@ -1,18 +1,115 @@
-import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { useQuery, useQueries, useMutation } from '@tanstack/react-query';
+import { Filter, applyFilters } from '@ovh-ux/manager-core-api';
+import { PaginationState } from '@ovh-ux/manager-react-components';
 import {
   checkPrivateNetworkCreationStatus,
   getPrivateNetworks,
   getSubnets,
+  deleteNetwork as apiDeleteNetwork,
 } from '@/data/api/networks';
-import { CreationStatus } from '@/types/network.type';
+import {
+  CreationStatus,
+  TGroupedSubnet,
+  TNetwork,
+  TSubnet,
+} from '@/types/network.type';
 import queryClient from '@/queryClient';
+import { groupedPrivateNetworkByVlanId, paginateResults } from '@/utils/utils';
+
+const networksQueryKey = (projectId: string, rest: string[] = []): string[] => [
+  'project',
+  projectId,
+  'network',
+  ...rest,
+];
 
 export const usePrivateNetworks = (projectId: string) =>
   useQuery({
-    queryKey: ['aggregated-network', projectId],
+    queryKey: networksQueryKey(projectId),
     queryFn: () => getPrivateNetworks(projectId),
   });
 
+export const useRegionPrivateNetworks = (
+  projectId: string,
+  pagination: PaginationState,
+  filters: Filter[] = [],
+) => {
+  const queryKey = networksQueryKey(projectId);
+  const data = queryClient.getQueryData<TNetwork[]>(queryKey) || [];
+
+  return useMemo(
+    () =>
+      paginateResults(
+        applyFilters(groupedPrivateNetworkByVlanId(data), filters),
+        pagination,
+      ),
+    [data, filters, pagination],
+  );
+};
+
+const getFormattedSubnets = (
+  networkId: string,
+  name: string,
+  region: string,
+  subnets: TSubnet[],
+): TGroupedSubnet[] =>
+  subnets.map(
+    ({
+      allocationPools,
+      id: subId,
+      cidr,
+      gatewayIp,
+      dhcpEnabled,
+      ipVersion,
+    }) => {
+      const allocatedIp = allocationPools
+        ?.map((i) => `${i.start} - ${i.end}`)
```

Comment:
Avoid using too generic property name or use destruct syntax
```suggestion
        ?.map((allocationPool) => `${allocationPool.start} - ${allocationPool.end}`)
```